keras/callbacks.py

```diff
@@ -381,3 +388,69 @@ def __init__(self, schedule):
 
     def on_epoch_begin(self, epoch, logs={}):
         self.model.optimizer.lr.set_value(self.schedule(epoch))
+
+
+class TensorBoard(Callback):
+    ''' Tensorboard basic visualizations.
+
+    This callback writes a log usable with tensorboard.
+
+    # Arguments
+        model: a keras model linked to a tensorflow session
+        feed: a dictionnary mapping tensors (inputs, outputs, weigths)
+            from the model._test keras function i.e. model._test.inputs
+            to the corresponding values
+        freq: the frequency at which the callback will output
+            parameters and metrics to the log
+        log_dir: the path of the directory where to save the log
+            files to be parsed by tensorboard
+    '''
+    def __init__(self, model, feed, freq=2, log_dir='./logs',
+                 show_accuracy=False):
+        super(Callback, self).__init__()
+        assert _BACKEND == 'tensorflow', \
+            'TensorBoard callback only works with the tensorflow backend'
+        import tensorflow as tf
+        import keras.backend.tensorflow_backend as tfbe
+
+        self.model = model
+        self.freq = freq
+        self.log_dir = log_dir
+        self.sess = tfbe._get_session()
+        self.feed = feed
+
+        mod_type = self.model.get_config()['name']
+        if mod_type == 'Sequential':
+            layers = {l.get_config()['name']: l for l in self.model.layers}
```

Comment:
I think this is a good solution for now. In the future when we have a layer naming system, we'll use that instead.